app/controllers/pages_controller.rb

```diff
@@ -52,6 +52,7 @@ def update
   private
 
   def render_liquid(layout)
+    localize_by_page_language(@page)
```

Comment:
this should come after the line that redirects to 404 if the page is inaccessible.